pkg/controller/common/test/controller-test-utils.go

```diff
@@ -5,18 +5,47 @@ import (
 	"fmt"
 	"testing"
 
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	clienttesting "k8s.io/client-go/testing"
-
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	clienttesting "k8s.io/client-go/testing"
+	"k8s.io/kubernetes/pkg/apis/rbac"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 
+	"github.com/maistra/istio-operator/pkg/apis"
 	"github.com/maistra/istio-operator/pkg/controller/common/test/assert"
 )
 
+func CreateClient(clientObjects ...runtime.Object) (client.Client, *EnhancedTracker) {
+	s := scheme.Scheme                          // scheme must be initialized before creating the client below
+	if err := apis.AddToScheme(s); err != nil { // scheme must be initialized before creating the client below
+		panic(fmt.Sprintf("Could not add to scheme: %v", err))
+	}
+	if err := rbac.AddToScheme(s); err != nil { // scheme must be initialized before creating the client below
+		panic(fmt.Sprintf("Could not add to scheme: %v", err))
+	}
+	s.AddKnownTypeWithName(schema.GroupVersionKind{
+		Group:   "k8s.cni.cncf.io",
+		Version: "v1",
+		Kind:    "NetworkAttachmentDefinition",
+	}, &unstructured.Unstructured{})
+	s.AddKnownTypeWithName(schema.GroupVersionKind{
+		Group:   "k8s.cni.cncf.io",
+		Version: "v1",
+		Kind:    "NetworkAttachmentDefinitionList",
+	}, &unstructured.UnstructuredList{})
+
+	tracker := clienttesting.NewObjectTracker(s, scheme.Codecs.UniversalDecoder())
```

Comment:
I think this should be initialized with a serializer created from the scheme, as opposed to the generic runtime scheme.

Reply:
I've changed this now. Can you see if it's what you meant?